File: LeetCode/Patterns/Two Pointers/167. Two Sum II - Input Array Is Sorted/Solution.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        n = len(numbers)

        left = 0
        right = n-1

        while left < right:
            l = numbers[left]
            r = numbers[right]

            if l + r == target:
                return [left+1, right+1]
            elif l + r < target:
                left += 1
            else:
                right -= 1
        
        return [-1, -1]

File: LeetCode/Patterns/Two Pointers/167. Two Sum II - Input Array Is Sorted/Solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int n = numbers.size();

        int left = 0;
        int right = n-1;

        while(left < right) {
            int l = numbers[left];
            int r = numbers[right];
            if(l + r == target) return {left+1, right+1};
            else if(l + r > target) right--;
            else left++;
        }

        return {-1, -1};
    }
};